ArchiSteamFarm/Steam/Data/InventoryDescription.cs

```diff
@@ -211,6 +211,51 @@ public EAssetRarity Rarity {
 
 								CachedRarity = EAssetRarity.Unknown;
 
+								return CachedRarity.Value;
+						}
+					case "RARITY":
+						switch (tag.internal_name) {
+							case "common" or "Rarity_common" or "Rarity_Common_Weapon" or "Rarity_Common_Character":
```

Comment:
It seems all `rarity` categories follow the same pattern: `Rarity_<rarity>*`, it might make more sense to split internal name on `_` then compare second part. Right now you're not only doing a lot of extra comparisons, you're also not bullet-proofing for the future if they e.g. add `Armor` to weapon and character tomorrow.